query/src/main/scala/filodb/query/exec/PeriodicSamplesMapper.scala

```diff
@@ -390,7 +397,8 @@ class SlidingWindowIterator(raw: RangeVectorCursor,
     * @param curWindowStart start time of the current window
     */
   private def shouldRemoveWindowHead(curWindowStart: Long): Boolean = {
-    (!windowQueue.isEmpty) && windowQueue.head.timestamp < curWindowStart
+    val windowStart = if (FiloQueryConfig.isInclusiveRange) curWindowStart else curWindowStart - 1
```

Comment:
Should this be `+1`? Same as on line 387?

Reply:
since we check if an element is falling out of the window, this should work